### PR TITLE
Use concept queries over suffix queries

### PR DESCRIPTION
**Pull request by 0ortmann**
###  :notebook_with_decorative_cover: Description

This PR depends on https://github.com/tenzir/vast/pull/1161

Rewrite suffix-based VAST queries to concept queries instead.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/threatbus](https://docs.tenzir.com/threatbus), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### CODE REVIEW

File: apps/vast/message_mapping.py [hunk 1]
```diff
@@ -78,14 +78,9 @@ def to_vast_query(intel: Intel):
...
-        # TODO: use field annotations, once implemented (ch17531)
-        return f'"{indicator}" in url'
+        return f'"{indicator}" in net.uri'
     if vast_type == "domain":
-        # Currently, uses VAST's suffix-based field matching. Targets every
-        # schema with fieldnames that end in `domain`, `host`, or `hostname`
-        # (Zeek and Suricata)
-        # TODO: use field annotations, once implemented (ch17531)
-        return f'"{indicator}" in domain || "{indicator}" in host || "{indicator}" in hostname'
+        return f'"{indicator}" in net.domain || "{indicator}" in net.hostname'
```
Comment (mavam) [inline apps/vast/message_mapping.py:83]: Just a general question here: how does that work with escaping? Can we have a unit test that checks that?

**Review by mavam — CHANGES_REQUESTED** [review]
Looks good, just one question regarding escaping of queries.

File: apps/vast/message_mapping.py (hunk 1 above)
Comment (0ortmann) [reply apps/vast/message_mapping.py:83]: What is the issue with escaping, can you give an example?

When you look at the two files that this PR touches, you'll see there already is a unit test.

Comment (mavam) [reply apps/vast/message_mapping.py:83]: Consider the indicator `e"il.com`. If you do not escape it properly, then the query expression is invalid. For such a corner case, it would make sense to have a unit test to ensure that this gets converted properly to `e\"il.com` before the interpolation takes place.

Comment (0ortmann) [reply apps/vast/message_mapping.py:83]: As discussed in the office, I created new story for this.

Comment (0ortmann) [reply apps/vast/message_mapping.py:83]: Review re-requested.

**Review by mavam — APPROVED** [review]
